charts/shoot-core/components/charts/coredns/templates/coredns-deployment.yaml

```
@@ -39,6 +39,8 @@ spec:
                       values:
                         - kube-dns
       priorityClassName: system-cluster-critical
+      nodeSelector:
+        system-components.gardener.cloud/allowed: true
```

```suggestion
        worker.gardener.cloud/system-components: true
```
"allowed" is functionally more related to taints/tolerations, i.e. if you create a system component pod w/o a label selector, nodes with `system-components.gardener.cloud/allowed: true` will also be considered.

---

Done